lms/templates/courseware/course_about.html

```diff
@@ -152,14 +153,34 @@
             reg_href = reg_then_add_to_cart_link
             reg_element_id = "reg_then_add_to_cart"
           %>
+          <% if ecommerce_checkout:
+              if user.is_authenticated():
```

Comment:
There should be no need to redirect to login/registration manually. Make sure the checkout view (both in LMS and Otto) requires login. Django will handle the rest.